addons/l10n_gcc_invoice/views/report_invoice.xml

```diff
@@ -344,27 +344,40 @@
                                     </td>
                                 </tr>
                                 <t t-set="tax_totals" t-value="o.tax_totals"/>
-                                <t t-foreach="tax_totals['subtotals']" t-as="subtotal">
-                                    <t t-set="subtotal_to_show" t-value="subtotal['name']"/>
+                                <t t-foreach="enumerate(tax_totals['subtotals'])" t-as="subtotal">
```

Comment:
the use the enumerate works fine, but I know there is another way (maybe a bit cleaner but quite unknown), which is to use the name of the variable followed by `_index`. Therefore you simply iterate over `tax_totals['subtotals']` and when the index is required, use `subtotal_index`. I believe it makes the code a bit cleaner and removes all the `[0]` and `[1]` you set. This comment also applies for the variable  `amount_by_group` that you set below

Reply:
Oh! I didn't know you could use `_index` like that. Out of curiosity, is there more documentation on how it works or do you have an explanation? Also, I implemented your suggestions